server/src/deployment/k18s/manager.ts

```diff
@@ -0,0 +1,219 @@
+
+    service.spec.ports = [v1ServicePort];
+
+    return this.k8sApi.createNamespacedService('default', service);
+  }
+
+  private async requestCreatePod(encryptedDisk: Disk, fingerprint: string) {
+    const volumeName = encryptedDisk.id();
+    const gceVolume = this.createGCEVolume(volumeName, encryptedDisk);
+    const container = this.createContainer(fingerprint, volumeName);
+
+    const v1Pod = new V1Pod();
+    v1Pod.kind = 'Pod';
+    v1Pod.apiVersion = 'v1';
+    v1Pod.spec = new V1PodSpec();
+    v1Pod.spec.volumes = [gceVolume];
+    v1Pod.spec.containers = [container];
+    v1Pod.metadata = new V1ObjectMeta();
+    v1Pod.metadata.name = 'pod-' + fingerprint;
+    v1Pod.metadata.labels = {};
+    v1Pod.metadata.labels['app'] = v1Pod.metadata.name;
+
+    return this.k8sApi.createNamespacedPod('default', v1Pod);
+  }
+
+  private createContainer(fingerprint: string, volumeName) {
+    const container = new V1Container();
+    // TODO: use SHA-1/commit hash based tagging?
+    container.image = 'gcr.io/arcs-project/deployment:latest';
+    container.name = 'container-image-' + fingerprint;
+    const volumeMount = this.createVolumeMount(volumeName);
+    container.volumeMounts = [volumeMount];
+
+    const v1ContainerPort = new V1ContainerPort();
+    v1ContainerPort.containerPort = 8080;
+    container.ports = [v1ContainerPort];
+    return container;
+  }
+
+  private createVolumeMount(volumeName) {
+    const volumeMount = new V1VolumeMount();
+    volumeMount.name = volumeName;
+    volumeMount.mountPath = '/personalcloud';
+    return volumeMount;
+  }
+
+  /**
+   * Create a reference to an existing GCE disk.
+   * @param volumeName the name of the volume
+   * @param encryptedDisk the existing disk already created by GCE
+   */
+  private createGCEVolume(volumeName, encryptedDisk: Disk) {
+    const gceVolume = new V1Volume();
+    gceVolume.name = volumeName;
+    gceVolume.gcePersistentDisk = new V1GCEPersistentDiskVolumeSource();
+    gceVolume.gcePersistentDisk.fsType = 'ext4';
+    gceVolume.gcePersistentDisk.pdName = encryptedDisk.id();
+    return gceVolume;
+  }
+
+  /**
+   * Creates a new Kubernetes 'PersistentVolume' object.
+   * @param k8sApi kubernetes API
+   * @param encryptedDisk the GCE disk to use
+   */
+  private async requestNewPersistentVolume(encryptedDisk: Disk) {
+    const newPersistentVolume = new V1PersistentVolume();
+    newPersistentVolume.metadata = new V1ObjectMeta();
+    newPersistentVolume.metadata.name = 'pv-' + encryptedDisk.id();
+    newPersistentVolume.apiVersion = 'v1';
+    newPersistentVolume.kind = 'PersistentVolume';
+    newPersistentVolume.spec = this.makePersistentVolumeSpec(encryptedDisk);
+    return this.k8sApi.createPersistentVolume(newPersistentVolume);
+  }
+
+  private makePersistentVolumeSpec(encryptedDisk: Disk) {
+    const spec = new V1PersistentVolumeSpec();
+    spec.accessModes = ["ReadWriteOnce"];
+    spec.capacity = {storage: '10Gi'};
```

Comment:
It appears to be required. The GCP disk has a specified size, but the Kubernetes API seems to want you to specify it as well even though it could discover it automatically.